src/components/order-service/order-service.controller.ts

```diff
@@ -183,22 +337,21 @@ export class OrderServiceController implements angular.IController {
     return serviceInstance;
   }
 
-  private gotoStepID(id: string) {
-    let step = _.find(this.ctrl.steps, { id: id });
-    this.gotoStep(step);
-  }
-
   private isNewProject(): boolean {
-    return this.ctrl.selectedProject && !_.has(this.ctrl.selectedProject, 'metadata.uid');
+    return !this.ctrl.selectedProject || !_.has(this.ctrl.selectedProject, 'metadata.uid');
   }
 
   private watchResults = (resource: any, data: any, context: any) => {
     this.watches.push(this.DataService.watchObject(resource, data.metadata.name, context, (instanceData: any, action: any) => {
       var conditions: any = _.get(instanceData, 'status.conditions');
       var readyCondition: any = _.find(conditions, {type: "Ready"});
 
+      this.ctrl.serviceInstanceName = _.get(instanceData, 'metadata.name');
       this.ctrl.orderComplete = readyCondition && readyCondition.status === 'True';
       this.ctrl.error = _.find(conditions, {type: "ProvisionFailed"});
+      if (this.ctrl.shouldBindToApp !== 'none' && !this.ctrl.bindInProgress) {
```

Comment:
You should have the data you need. The success handler to DataService.create gives you the result of the creation (the data arg), which is the updated serviceInstance object that contains the name